cmd/goose/main.go

```diff
@@ -34,17 +43,49 @@ func main() {
 		return
 	}
 
+	if os.Getenv(envKeyDriver) != "" || os.Getenv(envKeyDbstring) != "" {
+		return environment()
+	}
+
 	if len(args) < 3 {
-		flags.Usage()
+		help = true
 		return
 	}
 
-	if args[0] == "-h" || args[0] == "--help" {
-		flags.Usage()
+	driver, dbstring, command = args[0], args[1], args[2]
+	if len(args) > 3 {
+		arguments = append(arguments, args[3:]...)
+	}
+
+	return
+}
+
+func environment() (driver, dbstring, command string, arguments []string, help bool) {
+	var args []string
+
+	driver = os.Getenv(envKeyDriver)
+	dbstring = os.Getenv(envKeyDbstring)
+	args = flags.Args()
+	if len(args) < 1 {
+		help = true
 		return
 	}
+	command = args[0]
+	if len(args) > 1 {
+		arguments = append(arguments, args[1:]...)
+	}
 
-	driver, dbstring, command := args[0], args[1], args[2]
+	return
+}
+
+func main() {
+	driver, dbstring, command, arguments, help := configuration()
```

Comment:
I made changes respecting in style of the original.
In the original:
```driver, dbstring, command := args[0], args[1], args[2]```
I left all variables as is

`create` is also taken from the original:
https://github.com/pressly/goose/blob/master/cmd/goose/main.go#L30-L35

In fact, I replaced this part of the code (https://github.com/pressly/goose/blob/master/cmd/goose/main.go#L25-L48) with two functions. Everything else as is in original.

You can fix the code as you like, I do not mind.